AlarmClockSpeedTest.cpp

```diff
@@ -0,0 +1,88 @@
+/* 
+ * File:   AlarmClockSpeedTest.cpp
+ * Author: Amanda Carbonari
+ * Created: January 6, 2015 9:00am
+ */
+
+#include <ctime>
+#include <iostream>
+#include <AlarmClock.h> // To test other implementations, change this to #include<header file> or #inlcude<AlarmClock.h.old> for example
+#include <chrono>
+#include <thread>
+using namespace std;
+using namespace std::chrono;
+typedef std::chrono::microseconds microseconds;
+typedef std::chrono::milliseconds milliseconds;
+typedef std::chrono::seconds seconds;
+
+
+template<typename T> void WaitForAlarmClockToExpire(AlarmClock<T>& alerter) {
+   while(!alerter.Expired());
+}
+
+template <typename T> void testReset(unsigned int sleep_time, unsigned int reset_portion=2) {
```

Comment:
I'd also like to see a variant of the "for loop test". Basically, put a counter in a for loop that increments while ```!timer.Expired()``` and then print out the value of the counter for the old AlarmClock, the new AlarmClock, and for the StopWatch. This will show us how much extra compute time we get using one vs. the other.